### PR TITLE
Issues 1050

### PR DESCRIPTION
**Pull request by raftaar1191**
Description
This PR is for #1050

How Has This Been Tested?
Tested for all the display option that can be set from the form display tab one by one.
Have also tested the JS that it should not get a break with the current functionality.

## Types of changes
New feature (non-breaking change which adds functionality) 

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style.
- [x] My code follows has proper inline documentation.

### CODE REVIEW

**Comment by DevinWalker** [comment]
@raftaar1191 is this ready for review?

**Comment by raftaar1191** [comment]
Yes

File: includes/ajax-functions.php [hunk 1]
```diff
@@ -192,35 +192,39 @@ function give_ajax_get_form_title() {
...
-
-	echo $response;
-
-	give_die();
+	$response = array(
+		'success' => true,
+		'states_found' => $states_found,
+		'data' => $data,
+	);
+	wp_send_json( $response );
```
Comment (ravinderk) [inline includes/ajax-functions.php]: @raftaar1191 any reason not to use `wp_send_json_success` function

Comment (ravinderk) [inline includes/ajax-functions.php]: @raftaar1191 I agree as we discuss on call. As per current solution, state field will not print as text field as it was happening before. We can add a filter here to show/hide text field on basis of state if admin wants by default we can hide text field for `Germany`, `UAE` etc.

Comment (raftaar1191) [inline includes/ajax-functions.php]: wp_send_json_success internalty used the same function.
Here there is no possibility of wp_send_json_success or wp_send_json_error so I used wp_send_json in whcih parameter I can definne if it successfully or not

https://core.trac.wordpress.org/browser/tags/4.8/src/wp-includes/functions.php#L0

**Comment by DevinWalker** [comment]
@raftaar1191 I'm seeing an issue with the functionality with the state field being removed completely. See here when I make a change from the US to the United Kingdom:



The field is removed incorrectly. The functionality should work like WooCommerce does:

**Comment by raftaar1191** [comment]
@DevinWalker 

We can add a filter here to show/hide text field on basis of state if admin wants by default we can hide text field for Germany etc.

**Comment by DevinWalker** [comment]
@raftaar1191 definitely. Though, by default it should work like Woo does and show the text field for state and depending on the country have it required or not.

**Comment by raftaar1191** [comment]
@DevinWalker @ravinderk 

Done, now is working as in WooCommerces

**Comment by DevinWalker** [comment]
@raftaar1191 I'm still noticing differences. For instance, for Argentina WooCommerce loads the states whereas ours does not. Please ensure the states are loaded properly for all countries.

**Comment by raftaar1191** [comment]
@DevinWalker 

A States can be easily added to the country by using the `give_states_list` filter 
<a href="https://github.com/WordImpress/Give/pull/1858/files#diff-060a5c321b244e3d4a8903b351323c75R372" >FIlter link </a>

Was not know that need to add new states for the rest country as well.

Adding that now.

**Comment by DevinWalker** [comment]
This is working well now. Thanks @raftaar1191 !

**Pull request merged**